src/public/modules/forms/services/submissions.client.factory.js

```diff
@@ -93,18 +93,20 @@ function SubmissionsFactory(
     xhr.send(fd)
     // On Response
     xhr.onreadystatechange = function () {
-      // 4 DONE  The operation is complete.
       const OPERATION_DONE = 4
       if (xhr.readyState === OPERATION_DONE) {
         // waterfall is successful
+        let response = {}
+        try {
+          response = JSON.parse(xhr.responseText)
+          // eslint-disable-next-line no-empty
+        } catch (e) {}
         if (xhr.status === HttpStatus.OK) {
-          deferred.resolve('Submission has finished.')
+          deferred.resolve({
+            message: 'Submission has finished.',
+            id: response.submissionId || 'Not available',
```

Comment:
I don't think we need the `|| 'Not available'` since the API shape should always be consistent